src/structs/types/cache_information.rs

```diff
@@ -147,6 +155,37 @@ impl Serialize for SMBiosCacheInformation<'_> {
     }
 }
 
+/// # Maximum memory capacity, in kilobytes, for this cache item
+#[derive(Serialize, Debug, PartialEq, Eq)]
+pub enum CacheMemorySize {
+    /// Maximum memory capacity in Kilobytes
+    Kilobytes(u64),
+    /// Use `maximum_cache_size_2` to retrieve the maximum capacity
+    SeeMaximumCacheSize2,
+}
+
+impl From<u16> for CacheMemorySize {
```

Comment:
Comments for the review:
Maximum Cache Size and Install Size are both size WORD (u16) and follow the same logic:

Maximum size that can be installed
Bit 15 Granularity
0 – 1K granularity
1 – 64K granularity
Bits 14:0 Max size in given granularity
See 7.8.1.

For Cache sizes greater than 2047 MB, the
Maximum Cache Size field is set to 0xFFFF
and the Maximum Cache Size 2 field is
present, the Granularity bit is set to 1b, and
the size set as required; see 7.8.1.